examples/calendar/frontend/src/Appointment/Appointment.tsx

```diff
@@ -0,0 +1,219 @@
+import "./Appointment.css";
+
+import jwtDecode from "jwt-decode";
+import React, { useState, useEffect } from "react";
+
+import { Auth, User } from "../types";
+import axios from "axios";
+import {ToggleButton, Button, ButtonGroup} from 'react-bootstrap';
+import { useParams } from "react-router-dom";
+import { create } from "domain";
+interface Props {
+	auth: Auth;
+	onLoggedOut: () => void;
+}
+interface booking {
+    display: string;
+    value: string;
+}
+interface Time {
+	startTime: Date;
+	checked: boolean;
+	disabled: boolean;
+}
+interface SetBooking {
+	timeStarts: number[];
+	timeEnds: number[];
+	status: string;
+	userId: number;
+}
+interface State {
+	loading: boolean;
+	user?: {
+		id: number;
+		username: string;
+	};
+	username: string;
```

Comment:
[Question] We have `user.username` Why do we need `username` again?

Reply:
Removed one instance of username. Updated the code.